src/Hl7.Fhir.STU3/Specification/Source/Summary/ArtifactSummaryExtensions.cs

```diff
@@ -248,6 +248,42 @@ static TSource SingleOrDefault<TSource>(this IEnumerable<TSource> source, Func<I
             throw createException(source);
         }
 
+        /// <summary>
+        /// Determines if a resource version matches a query version according to FHIR canonical matching rules.
+        /// Supports both exact matching and partial version matching (e.g., "1.5" matches "1.5.0").
+        /// </summary>
+        /// <param name="resourceVersion">The version of the resource being checked.</param>
+        /// <param name="queryVersion">The version specified in the canonical URL query.</param>
+        /// <returns>True if the resource version matches the query version according to FHIR canonical matching rules.</returns>
+        private static bool MatchesVersion(string resourceVersion, string queryVersion)
```

Comment:
This logic is repeated three times in this PR. I think it would be better to unify it and put it, as a public function on a central place, e.g. on Canonical.